tests/app/questionnaire/test_questionnaire_store_updater.py

```diff
@@ -1404,10 +1429,180 @@ def test_dependent_sections_added_dependant_block_removed(
+
+    progress_store = ProgressStore(
+        [
+            {
+                "section_id": "section-1",
+                "block_ids": ["block-1", "block-2"],
+                "status": "IN PROGRESS",
+            },
+            {
+                "section_id": "section-2",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-3",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-4",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-5",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-6",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+            {
+                "section_id": "section-7",
+                "block_ids": ["block-1", "block-2"],
+                "status": "COMPLETED",
+            },
+        ],
+    )
+
+    questionnaire_store_updater = get_questionnaire_store_updater(
+        current_location=current_location,
+        progress_store=progress_store,
+        router=mock_router,
+        schema=mock_schema,
+    )
+
+    mocker.patch(
+        "app.questionnaire.questionnaire_store_updater.QuestionnaireStoreUpdater.get_chronological_section_dependents",
+        return_value=[
+            DependentSection(
+                section_id="section-1", list_item_id=None, is_complete=True
+            ),
+            DependentSection(
+                section_id="section-2", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-3", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-4", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-5", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-6", list_item_id=None, is_complete=None
+            ),
+            DependentSection(
+                section_id="section-7", list_item_id=None, is_complete=None
+            ),
+        ],
+    )
+
+    questionnaire_store_updater.update_progress_for_dependent_sections()
+
+    assert len(questionnaire_store_updater.evaluated_dependents) == 6
+
+
+def test_questionnaire_store_updater_dependency_evaluation_order(mocker, mock_router):
+    current_location = Location(section_id="section-2", block_id="list-collector")
+
+    schema = load_schema_from_name(
+        "test_progress_value_source_calculated_summary_extended"
+    )
```

Comment:
Does this schema have multiple chained deps, I suggest mocking the example one which has chained deps, like the one in the above test. It is easier to what is happening and we know the chained deps are tested then.

Reply:
No, but I just thought it was a good example displaying the ordering of dependency evaluation for chained deps even if not multiple

Reply:
Based on the fix to the other test I did, to check whether this is working as intended, I'm not sure if it is. What is the scenario we are testing for this and is it different to the other one?.

Reply:
I will remove this now given we have the updated parameterised test below 👍